firestore.rules

```diff
@@ -60,8 +65,15 @@ service cloud.firestore {
 
     match /campaigns/{campaignId} {
       allow read: if true;
-      allow create, update: if request.auth.uid != null;
-      
+
+      // Only authenticated users can create a campaign and only one campaign
+      allow create: if signedIn()
+                    && currentUser().data.createdCampaignId == null;
```

Comment:
I think this had issues too.. this will resolve to `auth.uid.data`

Reply:
These are all fixed in the 384-firebase-db-rules PR. I removed calling `.uid` in `currentUser()`. If we merge that PR, we can probably just cancel this PR.